### PR TITLE
Move class-level classes to package-level from Resources

### PR DESCRIPTION
**Pull request by dvoytenko**
No real refactoring here yet: just moved `Resource`, `TaskQueue` and `fixIeMediaBug` to separate modules and the same is done for tests.

### CODE REVIEW

File: src/service/resources-impl.js [hunk 1]
```diff
@@ -1070,6 +1036,9 @@ export class Resources {
         task.promise.then(this.taskComplete_.bind(this, task, true),
             this.taskComplete_.bind(this, task, false))
             .catch(reportError);
+        if (task.resource.getState() == ResourceState.LAYOUT_SCHEDULED) {
```
Comment (jridgewell) [inline src/service/resources-impl.js]: Where'd this come from?

**Comment by jridgewell** [comment]
LGTM.

File: src/service/resources-impl.js (hunk 1 above)
Comment (dvoytenko) [inline src/service/resources-impl.js]: It was done in the `Resource` class by calling internal `Resources` state. So, I decided to move it here - it's the same signal, just less leaking of `Resources` vars.

Comment (jridgewell) [inline src/service/resources-impl.js]: 👍

**Comment by cramforce** [comment]
MERGE MERGE MERGE!

Glad this is finally happening!

**Pull request merged**